import-scripts/annotation_suite/standardize_mutation_data.py

```diff
@@ -0,0 +1,1414 @@
+
+
+def extract_maf_data_from_file(filename, center_name, sequence_source):
+    rejected_vars_by_varclass = 0
+    rejected_vars_by_mutstatus = 0
+    rejected_vars_by_inc_data = 0
+    records_loaded = 0
+
+    header = get_file_header(filename)
+    data_file = open(filename, "r")
+    data_reader = [line for line in data_file.readlines() if not line.startswith("#")][1:]
+    maf_data = []
+    print("\nLoading data from file: %s" % (filename))
+    for line in data_reader:
+        add_record = True
+        if list(map(str.strip, line.split("\t"))) == header:
+            print("Double header in file! Skipping this line")
+            continue
+        data = dict(zip(header, list(map(str.strip, line.split("\t")))))
+        maf_record = create_maf_record_from_maf(data, center_name, sequence_source)
+        maf_data.append(maf_record)
+        records_loaded += 1
+    data_file.close()
+    print_data_loading_summary(filename, records_loaded, rejected_vars_by_varclass, rejected_vars_by_mutstatus, rejected_vars_by_inc_data, [])
+    return maf_data
+
+
+def print_data_loading_summary(filename, records_loaded, rejected_vars_by_varclass, rejected_vars_by_mutstatus, rejected_vars_by_inc_data, vcf_data_errors):
+    print("\tTotal records loaded %s " % (records_loaded))
+    if rejected_vars_by_varclass > 0:
+        print("\tTotal records filtered by variant classification: %s" % (rejected_vars_by_varclass))
+    if rejected_vars_by_mutstatus > 0:
+        print("\tTotal records rejected due to LOH or Wildtype Mutation Status: %s" % (rejected_vars_by_mutstatus))
+    if rejected_vars_by_inc_data > 0:
+        print("\tTotal records rejected due to incomplete allele data: %s" % (rejected_vars_by_inc_data))
+    if len(vcf_data_errors) > 0:
+        print("\tNumber of VCR data errors: %s" % (len(vcf_data_errors)))
+
+def write_standardized_mutation_file(maf_data, output_filename):
+    output_file = open(output_filename, "w")
+    output_file.write("\t".join(MAF_HEADER))
+    for data in maf_data:
+        formatted_data = list(map(lambda x: process_datum(data.get(x, "")), MAF_HEADER))
+        output_file.write("\n" + "\t".join(formatted_data))
+    output_file.write("\n")
+    output_file.close()
+    print("Standardized MAF written to: %s" % (output_filename))
+
+
+def generate_maf_from_input_data(input_directory, output_directory, extensions, center_name, sequence_source, has_vcf_data):
+    print("\nLoading data from input directory: %s" % (input_directory))
+    print("\n\tSearching for files with extensions: %s " % (", ".join(extensions)))
+
+    for filename in os.listdir(input_directory):
+        extract_data = False
+        for ext in extensions:
+            if filename.endswith(ext) and "normal" not in filename:
+                extract_data = True
+                break
+        if extract_data:
+            output_filename = os.path.join(output_directory, filename + ".temp")
+            if has_vcf_data and ".vcf" in filename:
```

Comment:
rename : has_vcf_data -> processing_vcf_data (because ".vcf" may have been in the extensions list but no .vcf files are present for a center)
Also, I wonder if this is necessary .. because in the setting of extract_data above, we look for valid matching extensions ... so I wonder how we might have set extract_data to true, but not be processing files of type vcf?

Reply:
removing `has_vcf_data`